orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -318,6 +318,9 @@ def backward(ctx, *grad_outputs):
         self._original_module_output_schema = None
         self._onnx_graphs_info = None
 
+        # Graph transformer config
+        self._propagate_cast_ops = False
```

Comment:
For now, this can be a binary switch. 
but we probably want to make this into a multi-level switch, something like O1, O2, ...